include/modem/lte_lc.h

```diff
@@ -204,9 +207,158 @@ enum lte_lc_pdp_type {
+	 *
+	 * CE levels 0-2 are supported in NB-IoT.
+	 * CE levels 0-1 are supported in LTE-M.
+	 */
+	enum lte_lc_ce_level ce_level;
+
+	/* EARFCN for given cell where EARFCN is per 3GPP TS 36.101. */
+	int earfcn;
+
+	/* Reduction in power density in dB. */
+	int16_t dl_pathloss;
+
+	/* Current RSRP level at time of report.
+	 * -17:  RSRP < -156 dBm
+	 * -16: -156 ≤ RSRP < -155 dBm
+	 *  ...
+	 * -3:  -143 ≤ RSRP < -142 dBm
+	 * -2:  -142 ≤ RSRP < -141 dBm
+	 * -1:  -141 ≤ RSRP < -140 dBm
+	 *  0:   RSRP < -140 dBm
+	 *  1:  -140 ≤ RSRP < -139 dBm
+	 *  2:  -139 ≤ RSRP < -138 dBm
+	 *  ...
+	 *  95: -46 ≤ RSRP < -45 dBm
+	 *  96: -45 ≤ RSRP < -44 dBm
+	 *  97: -44 ≤ RSRP dBm
+	 *  255: not known or not detectable
+	 */
+	int16_t rsrp;
+
+	/* Current RSRQ level at time of report.
+	 * -30:  RSRQ < -34 dB
+	 * -29: -34 ≤ RSRQ < -33.5 dB
+	 * ...
+	 * -2:  -20.5 ≤ RSRQ < -20 dB
+	 * -1:  -20 ≤ RSRQ < -19.5 dB
+	 *  0:   RSRQ < -19.5 dB
+	 *  1:  -19.5 ≤ RSRQ < -19 dB
+	 *  2:  -19 ≤ RSRQ < -18.5 dB
+	 *  ...
+	 *  32: -4 ≤ RSRQ < -3.5 dB
+	 *  33: -3.5 ≤ RSRQ < -3 dB
+	 *  34: -3 ≤ RSRQ dB
+	 *  35: -3 ≤ RSRQ < -2.5 dB
+	 *  36: -2.5 ≤ RSRQ < -2 dB
+	 *  ...
+	 *  45:  2 ≤ RSRQ < 2.5 dB
+	 *  46:  2.5 ≤ RSRQ dB
+	 *  255: not known or not detectable.
+	 */
+	int16_t rsrq;
+
+	/* Estimate of TX repetitions depending on RRC state measured in. 3GPP TS 36.331 and 36.213.
+	 *
+	 * RRC IDLE: Initial preamble repetition level in (N)PRACH based on CE level and
+	 *	     network configuration.
+	 *
+	 * RRC CONNECTED: Latest physical data channel (N)PUSCH transmission repetition level.
+	 */
+	int16_t tx_rep;
+
+	/* Estimate of RX repetitions depending on RRC state measured in. 3GPP TS 36.331 and 36.213.
+	 *
+	 * RRC IDLE: Initial Random Access control channel (M/NPDCCH) repetition level based on
+	 *	     CE level and network configuration.
+	 *
+	 * RRC CONNECTED: Latest physical data channel (N)PDSCH reception repetition level.
+	 */
+	int16_t rx_rep;
+
+	/* Physical cell ID of evaluated cell. */
+	int16_t phy_cid;
+
+	/* Current band information. 0 when band information is not available. 3GPP TS 36.101. */
+	int16_t band;
+
+	/* Current signal-to-noise ratio at time of report.
+	 * 0:   SNR < -24 dB
+	 * 1:  -24 dB <= SNR < -23 dB
+	 * 2:  -23 dB <= SNR < -22 dB
+	 * ...
+	 * 47:  22 dB <= SNR < 23 dB
+	 * 48:  23 dB <= SNR < 24 dB
+	 * 49:  24 dB <= SNR
+	 * 127: Not known or not detectable.
+	 */
+	int16_t snr;
+
+	/* Estimate of TX power in dBm depending on RRC state measured in.
+	 * 3GPP TS 36.101 and 36.213.
+	 *
+	 * RRC IDLE: Estimated TX power level is for first preamble transmission in (N)PRACH).
+	 *
+	 * RRC CONNECTED: Latest physical data channel (N)PUSCH transmission power level.
+	 */
+	int16_t tx_power;
+
+	/* Four byte E-UTRAN cell ID in hexadecimal format. */
+	char cell_id[AT_CONEVAL_CELL_ID_STRING_LENGTH_MAX];
+
+	/* String of MCC and MNC values. */
+	char plmn[AT_CONEVAL_PLMN_STRING_LENGTH_MAX];
```

Comment:
In #4356, I chose to split into MCC and MCC and present as integers. That could be considered here as well.